x-pack/plugins/fleet/server/routes/agent/handlers.test.ts

```diff
@@ -39,4 +41,19 @@ describe('getAvailableVersionsHandler', () => {
       items: ['300.0.0', '8.1.0', '8.0.0', '7.17.0'],
     });
   });
+
+  it('should not strip -SNAPSHOT from kibana version', async () => {
```

Comment:
Could you just specify that this happens when not in serverless? it's going to be easier for future changes.